matching-notification-api/src/main/java/org/phenotips/matchingnotification/notification/internal/DefaultPatientMatchEmail.java

```diff
@@ -239,4 +242,43 @@ public String getSubjectPatientId()
     {
         return this.subjectPatient.getPatientId();
     }
+
+    private JSONArray getFeatureMatchesJSON(PatientMatch match)
+    {
+        // we reconstruct features from corresponding details saved with match
+        // because there features are already Reordered via DefaultPhenotypesMap
+        Set<? extends Feature> matchFeatures = readFeatures(match.getMatchedDetails());
+        Set<? extends Feature> refFeatures = readFeatures(match.getReferenceDetails());
+
+        // passing null in place of access because the user is admin,
+        // access not used in DefaultPatientPhenotypeSimilarityView
+        PatientPhenotypeSimilarityView featuresView =
+            new DefaultPatientPhenotypeSimilarityView(matchFeatures, refFeatures, null);
```

Comment:
This is an implementation detail from another module, so it should be hidden. Use a factory instead, even if all the factory does is instantiate and return this `DefaultPatientPhenotypeSimilarityView`.

Reply:
@sdumitriu addressed in the last commit, if I got the idea right (please check).